app/code/community/TIG/MyParcel2014/Model/Api/MyParcel.php

```diff
@@ -914,37 +917,15 @@ protected function _getOptionsData(TIG_MyParcel2014_Model_Shipment $myParcelShip
             'only_recipient'        => (int)$myParcelShipment->isHomeAddressOnly(),
             'signature'             => (int)$myParcelShipment->isSignatureOnReceipt(),
             'return'                => (int)$myParcelShipment->getReturnIfNoAnswer(),
-            'label_description' => $myParcelShipment->getOrder()->getIncrementId(),
+            'label_description'     => $order->getIncrementId(),
+            'age_check'             => (int)$addressValidation->hasAgeCheck($order->getStoreId()),
         );
 
         if ($checkoutData !== null) {
 
-            if (key_exists('time', $checkoutData) && key_exists('price_comment', $checkoutData['time'][0]) && $checkoutData['time'][0]['price_comment'] !== null) {
-                switch ($checkoutData['time'][0]['price_comment']) {
-                    case 'morning':
-                        $data['delivery_type'] = self::TYPE_MORNING;
-                        break;
-                    case 'standard':
-                        $data['delivery_type'] = self::TYPE_STANDARD;
-                        break;
-                    case 'avond':
-                        $data['delivery_type'] = self::TYPE_NIGHT;
-                        break;
-                }
-            } elseif (key_exists('price_comment', $checkoutData) && $checkoutData['price_comment'] !== null) {
-                switch ($checkoutData['price_comment']) {
-                    case 'retail':
-                        $data['delivery_type'] = self::TYPE_RETAIL;
-                        break;
-                    case 'retailexpress':
-                        $data['delivery_type'] = self::TYPE_RETAIL_EXPRESS;
-                        break;
-                }
-            }
+            $data = $helper->getDeliveryType( $checkoutData, $data, $order );
```

Comment:
```suggestion
            $data = $helper->getDeliveryType($checkoutData, $data, $order);
```
Goed dat dit nu een eigen functie heeft!

Reply:
Done